openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -655,6 +673,10 @@ In addition to the response parameters defined in [@!RFC6749], the Authorization
   
 Note: The `credential_identifiers` parameter cannot be used when the `scope` parameter is used in the Authorization Request to request issuance of a Credential.
 
+Additional Token Response parameters MAY be defined and used,
+as described in [@!RFC6749].
+The Wallet MUST ignore any unrecognized parameters.
```

Comment:
Does that include unknown data fields in an `openid_credential` within the `authorization_details` structure?

RFC 9396, Section 7.1 states that the AS may enrich `authorization_details` in the Token Response. Since we added extensibility for the `openid_credential` type, I think we should be clear on how the Wallet must treat a response with unknown data fields. We can add a normative statement here or under #(authorization-details. @pmhsfelix had the same concern [in another comment](https://github.com/openid/OpenID4VCI/pull/382#discussion_r1757151581) and suggested that the Wallet MUST ignore any unrecognized data fields.

Reply:
I'm not sure what you mean by "unknown data fields" or where you'd like them to be ignored when they occur.  Can you maybe give an example?

FYI, I believe I addressed [@pmhsfelix's comment](https://github.com/openid/OpenID4VCI/pull/382#discussion_r1757151581) in [this commit](https://github.com/openid/OpenID4VCI/pull/382/commits/6659b29032ef359be0f87029366e76f45907420a) in which I corrected the party that must ignore the not-understood parameter.

Reply:
`authorization_details` can be used in *two* directions:
- On an authorization request, produced by the Wallet and consumed by the AS.
- On a token response, produced by the AS and consumed by the Wallet.
The later usage of `authorization_details` doesn't seem to be covered by the PR, i.e., there isn't any mention to the Wallet ignoring extra fields in the `authorization_details` produced by the AS and present in the *token response*.

Detail: regarding the use of `authorization_details` in the authorization request, should it be the *Credential Issuer* or the *Authorization Server* that must ignore them.

Reply:
If you have better text in mind @pmhsfelix please make a change suggestion that I can apply to the PR.  Thanks.

Reply:
@selfissued 
I can try to elaborate more on the issue. Using RAR, an Authorization Server MUST return `authorization_details` in the Token Response (see [RFC 9396 Section 7](https://datatracker.ietf.org/doc/html/rfc9396#name-token-response)). The `authorization_details` in the Token Response are not necessarily the same as in the Authorization or Token Request (see [RFC 9396 Section 7.1](https://datatracker.ietf.org/doc/html/rfc9396#name-enriched-authorization-deta)). Technically, with the extension mechanism for the `openid_credential` type, there could be a discrepancy between the Wallet and the Authorization Server, where the Authorization Server enriches an `openid_credential` type in the `authorization_details` array of a Token Response with extra fields that the Wallet does not understand (RFC 9396 assumes that all possible fields of a type are known upfront, which is not the case here). To avoid any undefined behavior, I suggest, that the Wallet must ignore any unknown fields of an `openid_credential` type in an `authorization_details` array in a Token Response.

Reply:
To be clear, the "unrecognized parameters" that the Wallet must ignore are not only parameters of the Token Response (as defined in RFC 6749) but also unrecognized data fields of the `openid_credential` type defined in this spec. The former is already defined, and the letter should also be defined imo.